pkg/flow/flow.go

```diff
@@ -215,9 +218,73 @@ func (c *Flow) LoadFile(f *File) error {
 	return diags.ErrorOrNil()
 }
 
+// ComponentInfos returns the component infos.
+func (c *Flow) ComponentInfos() []*ComponentInfo {
+	c.loadMut.RLock()
+	defer c.loadMut.RUnlock()
+
+	cns := c.loader.Components()
+	infos := make([]*ComponentInfo, len(cns))
+	edges := c.loader.OriginalGraph().Edges()
+	for i, com := range cns {
+		nn := newFromNode(com, edges)
+		infos[i] = nn
+	}
+	return infos
+}
+
 // Close closes the controller and all running components.
 func (c *Flow) Close() error {
 	c.cancel()
 	<-c.exited
 	return c.sched.Close()
 }
+
+func newFromNode(cn *controller.ComponentNode, edges []dag.Edge) *ComponentInfo {
+	references := make([]string, 0)
+	referencedBy := make([]string, 0)
+	for _, e := range edges {
+		if e.From.NodeID() == cn.NodeID() {
+			references = append(references, e.To.NodeID())
+		} else if e.To.NodeID() == cn.NodeID() {
+			referencedBy = append(referencedBy, e.From.NodeID())
+		}
+	}
+	h := cn.CurrentHealth()
+	ci := &ComponentInfo{
+		Label:        cn.Label(),
+		ID:           cn.NodeID(),
+		Name:         cn.ComponentName(),
+		Type:         "block",
+		References:   references,
+		ReferencedBy: referencedBy,
+		Health: &ComponentHealth{
+			State:       h.Health.String(),
+			Message:     h.Message,
+			UpdatedTime: h.UpdateTime,
+		},
+	}
+	return ci
+}
+
+// ComponentInfo represents a component in river.
```

Comment:
The comment here feels a little weird. It's not representing a component in river, necessarily. Should this say "represents a component in Flow?"